solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -227,35 +226,31 @@ public void extractTerms(Set<Term> terms) {
      */
     public abstract class FeatureScorer extends Scorer {
 
-      protected String name;
-      private HashMap<String,Object> docInfo;
+      final protected String name;
+      private DocInfo docInfo;
 
       public FeatureScorer(Feature.FeatureWeight weight) {
         super(weight);
         name = weight.getName();
+        docInfo = null;
       }
 
       @Override
       public abstract float score() throws IOException;
 
-
       /**
        * Used to provide context from initial score steps to later reranking steps.
        */
-      public void setDocInfo(HashMap<String,Object> iDocInfo) {
-        docInfo = iDocInfo;
+      public void setDocInfo(DocInfo docInfo) {
+        this.docInfo = docInfo;
       }
 
-      public Object getDocParam(String key) {
-        return docInfo.get(key);
+      public float getOriginalDocScoreOrDefault(float def) {
```

Comment:
Maybe this class should just have a getDocInfo(), so callers can add stuff directly to it without going through duplicate wrapper functions